scvid/module/_probabilisticpca.py

```diff
@@ -66,28 +78,19 @@ def __init__(
         self.sigma = PyroParam(lambda: torch.tensor(s), constraint=constraints.positive)
 
         # guide parameters
-        if ppca_flavor == "marginalized":
-            pass
-        elif ppca_flavor == "diagonal_normal":
-            self.L_gk = PyroParam(
-                lambda: torch.randn((g_genes, k_components), generator=rng)
-            )
+        if ppca_flavor == "diagonal_normal":
+            M_inv_kk = torch.linalg.inv(self.M_kk)
+            z_scale_k_init = torch.sqrt(torch.diag(self.sigma**2 * M_inv_kk)).detach()
```

Comment:
I don't think taking the diagonal entries of a potentially correlated covariance matrix is necessarily a good initialization, but I don't have a better suggestion either :)  Here's the thing though:  if `W_kg` is initialized randomly and `G >> K` , then `M_kk` is diagonal to a decent approximation.

Reply:
So we're probably OK here.